beacon-chain/sync/subscriber.go

```diff
@@ -279,6 +331,18 @@ func (s *Service) subscribeDynamicWithSubnets(
 				if s.chainStarted.IsSet() && s.cfg.InitialSync.Syncing() {
 					continue
 				}
+				valid, err := isDigestValid(digest, genesis, genRoot)
+				if err != nil {
+					log.Error(err)
+					continue
+				}
+				if !valid {
+					log.Warnf("Attestation subnets with digest %#x are no longer valid, unsubscribing from all of them.", digest)
+					// Unsubscribes from all our current subnets.
+					s.reValidateSubscriptions(subscriptions, []uint64{}, topicFormat, digest)
+					ticker.Done()
+					return
+				}
```

Comment:
The log message and comment are the same as in another place in this file, but the logic is different. It does not feel right.

Reply:
The message and log are the same because it is essentially performing the same activity(unsubscribing). The only difference is how it handles static subnets vs dynamic subnets.